app/permissions_migrations.go

```diff
@@ -200,6 +216,48 @@ func getAddBotPermissionsMigration() permissionsMap {
 	}
 }
 
+func applyChannelManageDeleteToChannelUser() permissionsMap {
+	return permissionsMap{
+		permissionTransformation{
+			On:  permissionAnd(isRole(model.CHANNEL_USER_ROLE_ID), onOtherRole(model.TEAM_USER_ROLE_ID, permissionExists(PERMISSION_MANAGE_PRIVATE_CHANNEL_PROPERTIES))),
+			Add: []string{PERMISSION_MANAGE_PRIVATE_CHANNEL_PROPERTIES},
+		},
+		permissionTransformation{
+			On:  permissionAnd(isRole(model.CHANNEL_USER_ROLE_ID), onOtherRole(model.TEAM_USER_ROLE_ID, permissionExists(PERMISSION_DELETE_PRIVATE_CHANNEL))),
+			Add: []string{PERMISSION_DELETE_PRIVATE_CHANNEL},
+		},
+		permissionTransformation{
+			On:  permissionAnd(isRole(model.CHANNEL_USER_ROLE_ID), onOtherRole(model.TEAM_USER_ROLE_ID, permissionExists(PERMISSION_MANAGE_PUBLIC_CHANNEL_PROPERTIES))),
+			Add: []string{PERMISSION_MANAGE_PUBLIC_CHANNEL_PROPERTIES},
+		},
+		permissionTransformation{
+			On:  permissionAnd(isRole(model.CHANNEL_USER_ROLE_ID), onOtherRole(model.TEAM_USER_ROLE_ID, permissionExists(PERMISSION_DELETE_PUBLIC_CHANNEL))),
+			Add: []string{PERMISSION_DELETE_PUBLIC_CHANNEL},
+		},
+	}
+}
+
+func removeChannelManageDeleteFromTeamUser() permissionsMap {
+	return permissionsMap{
+		permissionTransformation{
+			On:     permissionAnd(isRole(model.TEAM_USER_ROLE_ID), permissionExists(PERMISSION_MANAGE_PRIVATE_CHANNEL_PROPERTIES)),
+			Remove: []string{PERMISSION_MANAGE_PRIVATE_CHANNEL_PROPERTIES},
+		},
+		permissionTransformation{
+			On:     permissionAnd(isRole(model.TEAM_USER_ROLE_ID), permissionExists(PERMISSION_DELETE_PRIVATE_CHANNEL)),
+			Remove: []string{model.PERMISSION_DELETE_PRIVATE_CHANNEL.Id},
+		},
+		permissionTransformation{
+			On:     permissionAnd(isRole(model.TEAM_USER_ROLE_ID), permissionExists(PERMISSION_MANAGE_PUBLIC_CHANNEL_PROPERTIES)),
+			Remove: []string{PERMISSION_MANAGE_PUBLIC_CHANNEL_PROPERTIES},
+		},
+		permissionTransformation{
+			On:     permissionAnd(isRole(model.TEAM_USER_ROLE_ID), permissionExists(PERMISSION_DELETE_PUBLIC_CHANNEL)),
+			Remove: []string{PERMISSION_DELETE_PUBLIC_CHANNEL},
+		},
+	}
+}
+
```

Comment:
@grundleborg Should `applyChannelManageDeleteToChannelUser` and `removeChannelManageDeleteFromTeamUser` applied to Team Scheme roles too?

Reply:
Did we have team schemes a year ago?

Reply:
To be more specific - I don't really remember the context of this PR given how long ago it was, but probably it is correct because it's fixing old legacy default permissions, so unlikely to effect newer custom schemes even if they existed at this time.

Reply:
Sounds good. I don't know the background of these changes but a recent change I made required specific team scheme role types be migrated, which put these on my radar. No further action required.